src/coreComponents/physicsSolvers/multiphysics/CoupledSolver.hpp

```diff
@@ -72,7 +72,31 @@ class CoupledSolver : public PhysicsSolverBase
   /// deleted move operator
   CoupledSolver & operator=( CoupledSolver && ) = delete;
 
+  template< typename T >
+  void throwSolversNotFound( std::ostringstream & errorMessage, string const & solverType )
+  {
+    string_array availableSolvers;
+
+    this->getParent().forSubGroups( [&]( Group & group )
```

Comment:
```suggestion
    this->getParent().forSubGroups< T >( [&]( T & group )
```
Then you don't have to dynamic cast.